### ISSUE TITLE
C++ Compilation Error: LazyStringOutputStream is not a member of google::protobuf::io

### ISSUE THREAD

**Issue by xRaznick** [google/protobuf#3795]
On the 3.4 release with the 3.4 protoc.exe compiler, everything is going well.
When I pulled the 3.4.1 commits, protoc.exe still succeeds, but the compilation fails (part of the output):
```
2>messages.pb.cc
2>...\protobuf\src\google\protobuf\wire_format_lite.h(863): error C4146: unary minus operator applied to unsigned type, result still unsigned
2>...\protobuf\src\google\protobuf\wire_format_lite.h(874): error C4146: unary minus operator applied to unsigned type, result still unsigned
2>...\config\generated\messages.pb.cc(268): error C2039: 'LazyStringOutputStream': is not a member of 'google::protobuf::io'
2>...\protobuf\src\google\protobuf\io\zero_copy_stream_impl_lite.h(61): note: see declaration of 'google::protobuf::io'
2>...\config\generated\messages.pb.cc(268): error C2065: 'LazyStringOutputStream': undeclared identifier
```
It seems like [this commit](https://github.com/google/protobuf/commit/13fd045dbb2b4dacea32be162a41d5a4b0d1802f#diff-a60de1b93772a55cfdbb3c31e626b9c6) removed LazyStringOutputStream, so why is protoc.exe still generating code with this class?

**Comment by pherl** [comment]
This commit is only merged to master, and will be released as part of 3.5.x.  You probably synced your code against master but use the pre-compiled 3.4.x protoc, which is not compatible.

I'd suggest to either stick to tagged refs (e.g. v3.4.0, v.3.4.1) or build the protoc from source.

Closing this for now. Please let me know if there's still issues.

**Comment by s-bear** [comment]
This just happened to me. Completely fresh install using `vcpkg install protobuf[zlib] protobuf[zlib]:x64-windows`, as described here: [https://github.com/google/protobuf/blob/master/src/README.md#c-installation---windows](https://github.com/google/protobuf/blob/master/src/README.md#c-installation---windows). Output of that command follows.

``` > .\vcpkg.exe install protobuf[zlib] protobuf[zlib]:x64-windows
The following packages will be built and installed:
    protobuf[core,zlib]:x86-windows
    protobuf[core,zlib]:x64-windows
  * zlib[core]:x86-windows
  * zlib[core]:x64-windows
Additional packages (*) will be modified to complete this operation.
Starting package 1/4: zlib:x86-windows
Building package zlib[core]:x86-windows...
A suitable version of cmake was not found (required v3.10.2). Downloading portable cmake v3.10.2...
A suitable version of git was not found (required v2.15.0). Downloading portable git v2.15.0...
-- CURRENT_INSTALLED_DIR=C:/Users/uqspowe6/source/repos/vcpkg/installed/x86-windows
-- DOWNLOADS=C:/Users/uqspowe6/source/repos/vcpkg/downloads
-- CURRENT_PACKAGES_DIR=C:/Users/uqspowe6/source/repos/vcpkg/packages/zlib_x86-windows
-- CURRENT_BUILDTREES_DIR=C:/Users/uqspowe6/source/repos/vcpkg/buildtrees/zlib
-- CURRENT_PORT_DIR=C:/Users/uqspowe6/source/repos/vcpkg/ports/zlib/.
-- Downloading http://www.zlib.net/zlib-1.2.11.tar.gz...
-- Downloading http://www.zlib.net/zlib-1.2.11.tar.gz... OK
-- Testing integrity of downloaded file...
-- Testing integrity of downloaded file... OK
-- Extracting source C:/Users/uqspowe6/source/repos/vcpkg/downloads/zlib1211.tar.gz
-- Extracting done
-- Applying patch C:/Users/uqspowe6/source/repos/vcpkg/ports/zlib/cmake_dont_build_more_than_needed.patch
-- Applying patch C:/Users/uqspowe6/source/repos/vcpkg/ports/zlib/cmake_dont_build_more_than_needed.patch done
-- Downloading https://github.com/ninja-build/ninja/releases/download/v1.8.2/ninja-win.zip...
-- Downloading https://github.com/ninja-build/ninja/releases/download/v1.8.2/ninja-win.zip... OK
-- Testing integrity of downloaded file...
-- Testing integrity of downloaded file... OK
-- Configuring x86-windows-rel
-- Configuring x86-windows-rel done
-- Configuring x86-windows-dbg
-- Configuring x86-windows-dbg done
-- Build x86-windows-rel
-- Build x86-windows-rel done
-- Build x86-windows-dbg
-- Build x86-windows-dbg done
-- Installing: C:/Users/uqspowe6/source/repos/vcpkg/packages/zlib_x86-windows/share/zlib/copyright
-- Performing post-build validation
-- Performing post-build validation done
Building package zlib[core]:x86-windows... done
Installing package zlib[core]:x86-windows...
Installing package zlib[core]:x86-windows... done
Elapsed time for package zlib:x86-windows: 3.296 min
Starting package 2/4: protobuf:x86-windows
Building package protobuf[core,zlib]:x86-windows...
-- CURRENT_INSTALLED_DIR=C:/Users/uqspowe6/source/repos/vcpkg/installed/x86-windows
-- DOWNLOADS=C:/Users/uqspowe6/source/repos/vcpkg/downloads
-- CURRENT_PACKAGES_DIR=C:/Users/uqspowe6/source/repos/vcpkg/packages/protobuf_x86-windows
-- CURRENT_BUILDTREES_DIR=C:/Users/uqspowe6/source/repos/vcpkg/buildtrees/protobuf
-- CURRENT_PORT_DIR=C:/Users/uqspowe6/source/repos/vcpkg/ports/protobuf/.
-- Downloading https://github.com/google/protobuf/releases/download/v3.5.1/protobuf-cpp-3.5.1.tar.gz...
-- Downloading https://github.com/google/protobuf/releases/download/v3.5.1/protobuf-cpp-3.5.1.tar.gz... OK
-- Testing integrity of downloaded file...
-- Testing integrity of downloaded file... OK
-- Downloading https://github.com/google/protobuf/releases/download/v3.5.1/protoc-3.5.1-win32.zip...
-- Downloading https://github.com/google/protobuf/releases/download/v3.5.1/protoc-3.5.1-win32.zip... OK
-- Testing integrity of downloaded file...
-- Testing integrity of downloaded file... OK
-- Extracting source C:/Users/uqspowe6/source/repos/vcpkg/downloads/protobuf-cpp-3.5.1.tar.gz
-- Extracting done
-- Applying patch C:/Users/uqspowe6/source/repos/vcpkg/ports/protobuf/001-add-compiler-flag.patch
-- Applying patch C:/Users/uqspowe6/source/repos/vcpkg/ports/protobuf/001-add-compiler-flag.patch done
-- Applying patch C:/Users/uqspowe6/source/repos/vcpkg/ports/protobuf/export-ParseGeneratorParameter.patch
-- Applying patch C:/Users/uqspowe6/source/repos/vcpkg/ports/protobuf/export-ParseGeneratorParameter.patch done
-- Extracting source C:/Users/uqspowe6/source/repos/vcpkg/downloads/protoc-3.5.1-win32.zip
-- Extracting done
-- Configuring x86-windows-rel
-- Configuring x86-windows-rel done
-- Configuring x86-windows-dbg
-- Configuring x86-windows-dbg done
-- Build x86-windows-rel
-- Build x86-windows-rel done
-- Build x86-windows-dbg
-- Build x86-windows-dbg done
-- Installing: C:/Users/uqspowe6/source/repos/vcpkg/packages/protobuf_x86-windows/share/protobuf/copyright
-- Installing: C:/Users/uqspowe6/source/repos/vcpkg/packages/protobuf_x86-windows/tools/protoc.exe
-- Performing post-build validation
-- Performing post-build validation done
Building package protobuf[core,zlib]:x86-windows... done
Installing package protobuf[core,zlib]:x86-windows...
Installing package protobuf[core,zlib]:x86-windows... done
Elapsed time for package protobuf:x86-windows: 4.015 min
Starting package 3/4: zlib:x64-windows
Building package zlib[core]:x64-windows...
-- CURRENT_INSTALLED_DIR=C:/Users/uqspowe6/source/repos/vcpkg/installed/x64-windows
-- DOWNLOADS=C:/Users/uqspowe6/source/repos/vcpkg/downloads
-- CURRENT_PACKAGES_DIR=C:/Users/uqspowe6/source/repos/vcpkg/packages/zlib_x64-windows
-- CURRENT_BUILDTREES_DIR=C:/Users/uqspowe6/source/repos/vcpkg/buildtrees/zlib
-- CURRENT_PORT_DIR=C:/Users/uqspowe6/source/repos/vcpkg/ports/zlib/.
-- Using cached C:/Users/uqspowe6/source/repos/vcpkg/downloads/zlib1211.tar.gz
-- Testing integrity of cached file...
-- Testing integrity of cached file... OK
-- Extracting done
-- Applying patch C:/Users/uqspowe6/source/repos/vcpkg/ports/zlib/cmake_dont_build_more_than_needed.patch
-- Applying patch failed. This is expected if this patch was previously applied.
-- Applying patch C:/Users/uqspowe6/source/repos/vcpkg/ports/zlib/cmake_dont_build_more_than_needed.patch done
-- Configuring x64-windows-rel
-- Configuring x64-windows-rel done
-- Configuring x64-windows-dbg
-- Configuring x64-windows-dbg done
-- Build x64-windows-rel
-- Build x64-windows-rel done
-- Build x64-windows-dbg
-- Build x64-windows-dbg done
-- Installing: C:/Users/uqspowe6/source/repos/vcpkg/packages/zlib_x64-windows/share/zlib/copyright
-- Performing post-build validation
-- Performing post-build validation done
Building package zlib[core]:x64-windows... done
Installing package zlib[core]:x64-windows...
Installing package zlib[core]:x64-windows... done
Elapsed time for package zlib:x64-windows: 45.17 s
Starting package 4/4: protobuf:x64-windows
Building package protobuf[core,zlib]:x64-windows...
-- CURRENT_INSTALLED_DIR=C:/Users/uqspowe6/source/repos/vcpkg/installed/x64-windows
-- DOWNLOADS=C:/Users/uqspowe6/source/repos/vcpkg/downloads
-- CURRENT_PACKAGES_DIR=C:/Users/uqspowe6/source/repos/vcpkg/packages/protobuf_x64-windows
-- CURRENT_BUILDTREES_DIR=C:/Users/uqspowe6/source/repos/vcpkg/buildtrees/protobuf
-- CURRENT_PORT_DIR=C:/Users/uqspowe6/source/repos/vcpkg/ports/protobuf/.
-- Using cached C:/Users/uqspowe6/source/repos/vcpkg/downloads/protobuf-cpp-3.5.1.tar.gz
-- Testing integrity of cached file...
-- Testing integrity of cached file... OK
-- Using cached C:/Users/uqspowe6/source/repos/vcpkg/downloads/protoc-3.5.1-win32.zip
-- Testing integrity of cached file...
-- Testing integrity of cached file... OK
-- Extracting done
-- Applying patch C:/Users/uqspowe6/source/repos/vcpkg/ports/protobuf/001-add-compiler-flag.patch
-- Applying patch failed. This is expected if this patch was previously applied.
-- Applying patch C:/Users/uqspowe6/source/repos/vcpkg/ports/protobuf/001-add-compiler-flag.patch done
-- Applying patch C:/Users/uqspowe6/source/repos/vcpkg/ports/protobuf/export-ParseGeneratorParameter.patch
-- Applying patch failed. This is expected if this patch was previously applied.
-- Applying patch C:/Users/uqspowe6/source/repos/vcpkg/ports/protobuf/export-ParseGeneratorParameter.patch done
-- Extracting done
-- Configuring x64-windows-rel
-- Configuring x64-windows-rel done
-- Configuring x64-windows-dbg
-- Configuring x64-windows-dbg done
-- Build x64-windows-rel
-- Build x64-windows-rel done
-- Build x64-windows-dbg
-- Build x64-windows-dbg done
-- Installing: C:/Users/uqspowe6/source/repos/vcpkg/packages/protobuf_x64-windows/share/protobuf/copyright
-- Installing: C:/Users/uqspowe6/source/repos/vcpkg/packages/protobuf_x64-windows/tools/protoc.exe
-- Performing post-build validation
-- Performing post-build validation done
Building package protobuf[core,zlib]:x64-windows... done
Installing package protobuf[core,zlib]:x64-windows...
Installing package protobuf[core,zlib]:x64-windows... done```
